src/Spd.Utilities.Dynamics/DynamicsContextLookupHelpers.cs

```diff
@@ -429,5 +429,16 @@ public static string LookupLicenceCategoryKey(Guid? licenceCategoryId)
                     throw;
             }
         }
+        public static async Task<contact?> GetDuplicateContact(this DynamicsContext context, contact contact, CancellationToken ct)
+        {
+            if (string.IsNullOrEmpty(contact.spd_bcdriverslicense))

```

Comment:
Emad, sorry, I just confirmed with Kenny, the duplicate check for contact should be
if there is driver licence, then compare them
if there is no driver licence, then just compare name and birthdate.
So, your original code is correct.

Reply:
yes I see it, so I'll change it